src/sha256_avx_x8.S

```diff
@@ -589,6 +589,9 @@ hashtree_sha256_avx2_x8:
         ret
 .Lstart_routine:       
 	push 	rbp
+#ifdef __WIN64__
+	push rsi
+#endif
```

Comment:
This is critical, checked that all other assembly implementations do preserve RSI correctly.